### ISSUE TITLE
Support registering custom action buttons in Moderation UI

### ISSUE THREAD

**Issue by ThisIsMissEm** [mastodon/mastodon#32284]
### Pitch

The moderation UI cannot be all things to all people, however, one way we can reduce friction and context switching is by adding in custom action buttons that are scoped to specific areas of the moderation UI

e.g., Say I want to have a custom action that looks up the email domain on an Account and returns back whether it's newly registered or not, this is obviously pretty specific to my server, but if I wanted to build this, I'd need to build a third-party tool like a discord bot or something with a complete accounts management UI, when really I just want to look up and return additional information.

With Custom Actions, the integration is essentially a webhook triggered by a button in the Account Show page. When pressed it creates a call record that's a state machine, the payload to the webhook contains the Account Entity and a URI to send updates back to.

My service then listens for this custom action & returns back either a failure, times out, or success. On success the payload contains an update to store (e.g., create an account note with this text)

On error or timeout (custom action invocations have a limited runtime) an error message is shown in the UI.

We already have all the machinery for webhooks, these just allow data to flow back into Mastodon on completion, so we'd need the custom action registration, the invocation records and small entry points from which the buttons can be rendered in the UI.

Areas I think could have them:
- Report overview
- Report details
- Account overview 
- Account details
- Instance details
- IP rules (maybe)
- Trends (maybe)

### Motivation

These would allow building deeper "native" integrations into the Mastodon Moderation UI without the team needing to build all these features into Mastodon.

**Comment by ThisIsMissEm** [comment]
We could also add an ID Token and Access Token into the payload to allow the remote endpoint to invoke the Mastodon API as that user

**Comment by renchap** [comment]
This should probably be something enabled by FASPs, here through trust & safety capabilities, so they can add their own buttons to the interface

**Comment by ThisIsMissEm** [comment]
@renchap yeah, it overlaps, but I think it'd need to have more of a registration flow in FASPs but would be good without all the FASP machinery too. Shipping a kiss version of this as an MVP seems smart
